engine/src/main/java/org/entando/entando/aps/system/services/tenants/TenantService.java

```diff
@@ -64,6 +70,48 @@ public List<TenantDto> getTenants() {
 
     }
 
+    @Override
+    public TenantStatsAndStatusesDto getTenantStatsAndStatuses() {
+        Map<String, TenantStatus> statuses = tenantManager.getStatuses();
+        Map<String, TenantStatus> mandatory = fetchByFilteringConfig(statuses, true);
+        Map<String, TenantStatus> additionals = fetchByFilteringConfig(statuses, false);
+
+        return TenantStatsAndStatusesDto.builder()
+                .mandatoryStatuses(mandatory)
+                .additionalStatuses(additionals)
+                .stats(calculate(statuses)).build();
+    }
+
+    private Map<String, TenantStatus> fetchByFilteringConfig(Map<String, TenantStatus> statuses, boolean isMandatory) {
+        return statuses.keySet().stream()
+                .flatMap(code -> tenantManager.getConfig(code).stream())
+                .filter(tc -> isMandatory == tc.isInitializationAtStartRequired())
+                .map(TenantConfig::getTenantCode)
+                .collect(Collectors.toMap(k -> k, statuses::get));
+    }
+
+    private TenantStatsDto calculate(Map<String, TenantStatus> statuses) {
+        TenantStatsDto d = new TenantStatsDto(0,0,0,0,0);
```

Comment:
Any reason here to not use `@NoArgsConstructor` and simply instantiate this using `new TenantStatsDto()`? Looking at this constructor it's not easy to understand which parameter populates which field.

Reply:
done